### PR TITLE
Add support for inline expect tests.

### PR DESCRIPTION
**Pull request by ezyang**
expecttest and test_expecttest are the implementation and tests
for this functionality.  I wired it up to the --accept flag,
but there's also a new environment variable EXPECTTEST_ACCEPT
which may be more convenient to trigger.  Haven't tested if this
works in fbcode.

There may be a few expect tests which will benefit from inline
treatment, but I just did one to show it works.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

**Review by ailzhang — APPROVED** [review]
Thanks!

File: test/test_expecttest.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+''')
+""")
+
+
+def load_tests(loader, tests, ignore):
+    tests.addTests(doctest.DocTestSuite(expecttest))
+    return tests
+
+
+if __name__ == '__main__':
```
Comment (SsnL) [inline test/test_expecttest.py]: Add this to `run_tests.py`?

**Comment by ezyang** [comment]
By the way, it seems reasonably easy to accidentally go over the line limit. I considered some options; the option I like best is to turn off line linting altogether: https://github.com/pytorch/pytorch/pull/12825/commits/296eb1e1ea5179663817c09961eb27af436af087#diff-b91f3d5bd63fcd17221b267e851608e8

But we can do more surgical things; for example, turn off line length linting for test files only.

**Comment by SsnL** [comment]
FWIW @ezyang I've done noqa on some conv and pooling docs where the formula is too long.

**Pull request closed without merging**